### PR TITLE
perf: alternative shape for lists

### PR DESCRIPTION
**Pull request by giacomociti**
for some reason, the current shapes for lists (and also those in shacl-shacl) cause perf issues to the [shacl library](https://github.com/zazuko/rdf-validate-shacl), which fails with big lists (in the order of the thousands elements).

For example, using this constraint:
```bash
$ barnard59 cube fetch-constraint --endpoint https:/int.lindas.admin.ch/query --cube https://energy.ld.admin.ch/elcom/electricityprice > constraint.ttl
```
and trying to validate it:
```bash
barnard59 shacl validate --shapes validation/standalone-constraint-constraint.ttl < constraint.ttl
```
we get the error: Maximum call stack size exceeded.

The error disappears with the shapes proposed in this PR.

I wanted to add an approval test but I gave up because, besides the main meaningful  validation messages, the report includes many confusing result details (the same happens also with the existing list shapes). 

This is a simplified [repro]( https://s.zazuko.com/iDGgdo) with an unexpected focusNode in the result, maybe related to a [known issue](https://github.com/zazuko/rdf-validate-shacl/issues/53)

### CODE REVIEW

**Comment by tpluscode** [comment]
cc @Rdataflow

**Review by tpluscode — CHANGES_REQUESTED** [review]
Please add a test case with some broken lists

**Comment by Rdataflow** [comment]
@giacomociti this allows to check very long lists 🎉 🎉 🎉

@tpluscode IIUC with this change we can bump the list length to ~ 5000 in cube-creator, correct? (... of course you'd add some CI tests to confirm it actually works as expected 👍)

**Comment by Rdataflow** [comment]
@giacomociti I observe some odd behaviour and results using 
```sh
barnard59 cube fetch-constraint --endpoint https:/test.lindas.admin.ch/query --cube https://energy.ld.admin.ch/elcom/electricityprice > constraint_test.ttl
barnard59 shacl validate --shapes validation/standalone-constraint-constraint.ttl < constraint_test.ttl | barnard59 shacl report-summary
```
resulting in 
```
warn: 13 results with severity http://www.w3.org/ns/shacl#Warning {"timestamp":"2024-03-27T10:05:49.952Z"}
error: 29 violations found {"stack":"Error: 29 violations found\n    at Object.<anonymous> (file:///C:/Users/Z70ATBE/node_modules/barnard59-shacl/lib/report.js:35:18)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async nextAsync (node:internal/streams/from:182:33)","timestamp":"2024-03-27T10:05:50.062Z"}
Warning of NodeConstraintComponent: "The use of qudt:unit is deprecated, please use qudt:hasUnit instead" with path <http://www.w3.org/ns/shacl#property> at focus node <https://energy.ld.admin.ch/elcom/electricityprice/shape> (source: _:b88_b630)
Violation of NodeConstraintComponent: "Value does not have shape <https://cube.link/shape/standalone-constraint-constraint#listnode>" with path null at focus node <http://www.w3.org/1999/02/22-rdf-syntax-ns#nil> (source: _:b88_b671)
Violation of MinCountConstraintComponent: "list node needs exactly one rdf:first" with path <http://www.w3.org/1999/02/22-rdf-syntax-ns#first> at focus node <http://www.w3.org/1999/02/22-rdf-syntax-ns#nil> (source: _:b88_b674)
Violation of MinCountConstraintComponent: "list node needs exactly one rdf:rest" with path <http://www.w3.org/1999/02/22-rdf-syntax-ns#rest> at focus node <http://www.w3.org/1999/02/22-rdf-syntax-ns#nil> (source: _:b88_b675)
Violation of EqualsConstraintComponent: "Must have same values as <http://qudt.org/schema/qudt/hasUnit>" with path <http://qudt.org/schema/qudt/unit> at focus node _:b88_b88_b88_genid2d5510cd0133f64e44a42fe88022dd4e2a2d0b4361 (source: _:b88_b631)
Warning of NodeConstraintComponent: "The use of qudt:unit is deprecated, please use qudt:hasUnit instead" with path <http://www.w3.org/ns/shacl#property> at focus node <https://energy.ld.admin.ch/elcom/electricityprice/shape> (source: _:b88_b630)
Violation of EqualsConstraintComponent: "Must have same values as <http://qudt.org/schema/qudt/hasUnit>" with path <http://qudt.org/schema/qudt/unit> at focus node _:b88_b88_b88_genid2d5510cd0133f64e44a42fe88022dd4e2a2d0b4362 (source: _:b88_b631)
Warning of NodeConstraintComponent: "The use of qudt:unit is deprecated, please use qudt:hasUnit instead" with path <http://www.w3.org/ns/shacl#property> at focus node <https://energy.ld.admin.ch/elcom/electricityprice/shape> (source: _:b88_b630)
Violation of EqualsConstraintComponent: "Must have same values as <http://qudt.org/schema/qudt/hasUnit>" with path <http://qudt.org/schema/qudt/unit> at focus node _:b88_b88_b88_genid2d5510cd0133f64e44a42fe88022dd4e2a2d0b4363 (source: _:b88_b631)
```
while obviously the validation fails due to missing `qudt:hasUnit` there are oddly 3 violations on lists ... which seem non-related but oddly disappear once `qudt:hasUnit` is manually added to the constraint_test.ttl
```
Violation of NodeConstraintComponent: "Value does not have shape <https://cube.link/shape/standalone-constraint-constraint#listnode>" with path null at focus node <http://www.w3.org/1999/02/22-rdf-syntax-ns#nil> (source: _:b88_b671)
Violation of MinCountConstraintComponent: "list node needs exactly one rdf:first" with path <http://www.w3.org/1999/02/22-rdf-syntax-ns#first> at focus node <http://www.w3.org/1999/02/22-rdf-syntax-ns#nil> (source: _:b88_b674)
Violation of MinCountConstraintComponent: "list node needs exactly one rdf:rest" with path <http://www.w3.org/1999/02/22-rdf-syntax-ns#rest> at focus node <http://www.w3.org/1999/02/22-rdf-syntax-ns#nil> (source: _:b88_b675)
```
thanks for having a look at this situation

**Comment by giacomociti** [comment]
@Rdataflow the strange behavior disappears using `sh:xone` (which is even more precise). We will open an issue to investigate the validator

**Review by tpluscode — APPROVED** [review]
Should be good here but we need to fix the out of place `sh:details`...

File: validation/standalone-constraint-constraint.ttl [hunk 1]
```diff
@@ -65,10 +65,16 @@
         ] ,
         [
             sh:path sh:property ;
-            sh:node [
-                sh:path qudt:unit ;
-                sh:equals qudt:hasUnit ;
-            ] ;
+            sh:or (
```
Comment (Rdataflow) [inline validation/standalone-constraint-constraint.ttl:68]: @giacomociti good catch - replace with `sh:xone` :100: 

see also https://github.com/zazuko/cube-link/pull/162

**Review by tpluscode — APPROVED** [review]

**Pull request merged**